src/adagrams.js

```diff
@@ -1,15 +1,100 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
-};
+  let alphabetPool = 'aaaaaaaaaabbccddddeeeeeeeeeeeeffggghhiiiiiiiiijkllllmmnnnnnnooooooooppqrrrrrrssssttttttuuuuvvwwxyyz'
+  alphabetPool = alphabetPool.toUpperCase()
+  let i = 0;
+  let playerHand = []
+  
+  do {
+    i++;
+    let singleLetter = alphabetPool.charAt(Math.random() * alphabetPool.length);
+    playerHand.push(singleLetter);
+    alphabetPool = alphabetPool.replace(singleLetter, '');
+  } while (i < 10);
+
+  return playerHand;
+}
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  let playerHand = {}
+  for (let i of lettersInHand) {
+    if (i in playerHand) {
+      playerHand[i] += 1
+    } else {
+      playerHand[i] = 1
+    }
+  } 
+
+  for (let j of input) {
+    if (!(j in playerHand)) {
+      return false;
+    } else {
+      playerHand[j] -= 1
+    }
+  }
+
+  return Object.values(playerHand).every(
+    value => value >= 0
+  )
```

Comment:
This works and looks readable, Well done. However You can do this in a more efficient manner by using one if statement that returns false if the user's letter is not in the letters in hand. You can also simply return true in the function and this would also work because the condion would only check for invalid words. Essentially you always return true if the user's input is valid, if not; we move into the conditional:

...
  // Loop through the letters in the user's input
  for (let i = 0; i < input.length; i++) {
    // If the user's letter is not in the letters in hand
    if (lettersInHandCopy.includes(input_array[i])) {
      // Remove the letter from the letters in the hand
      // The shift() method in JavaScript is used to remove the first element from an array and return that element.
      lettersInHandCopy.shift(input_array[i]);
    // If the user's letter is not in the letters in hand
    } else {
      return false;
    }
  }
  // If the user's input is valid
  return true;
};